### DISCUSSION TITLE
Plotting with matplotlib results in a wider than expected graphic

### DISCUSSION THREAD

**Discussion by mauri491** [lightkurve/lightkurve#1408] [Q&A]
As the title says, I'm having an issue when rendering pixelfile data, the resulting plot has an extra wide space to the left that doesn't seem to go away even with tight_layout() being called. What could be the issue?

```
import lightkurve as lk
import numpy as np
import matplotlib.pyplot as plt

tpf_example = lk.search_targetpixelfile("KIC 7461601", author="Kepler", cadence='long', quarter=10).download()
tpf_example.plot()
plt.tight_layout()
plt.show()
```

![image](https://github.com/lightkurve/lightkurve/assets/82736282/ac814988-fd4f-4fce-89d0-6569ed3c2b6b)

**Comment by orionlee** [comment]
The issue is related to matploblib's interactive widget (it looks like either `ipympl` or `notebook` is used, based on the screenshot).

If you use static output instead (`%matplotlib inline`), the extra whitespace should be gone.

---

As an illustration, even if one uses the barebone `plt.imshow` to plot the pixels, when using interactive `ipympl`, the output also has extra whitespace.

```python
%matplotlib ipympl

plt.imshow(tpf_example.flux[0].value, origin="lower")
plt.show()
```

In contrast, when using `%matplotlib inline` , the extra whitespace is gone.

**Reply by mauri491** [reply]
I know the problem isn't present when using static outputs, but I was hoping there'd be a way to overcome this, as I don't use notebook or ipyml. I am trying to save the images of multiple subplots automatically from running the code in Vscode but they also have the whitespace.

**Reply by orionlee** [reply]
I'd suggest you post the barebone problem (without lightkurve) to the forum of the specific matplotlib backend you're using. 
If I'm not mistaken, the whitespace is there even when barebone `plt.imshow()` over a 2-D array is used.

**Reply by orionlee** [reply]
BTW, have you tried `plt.savefig(filename, bbox_inches="tight")`, given you're trying to save the images from codes?

https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html
